Library/Homebrew/cmd/log.rb

```diff
@@ -14,14 +14,24 @@ def log
 
   private
 
-  def git_log(path=nil)
+  def git_log(path = nil)
     if File.exist? "#{`git rev-parse --show-toplevel`.chomp}/.git/shallow"
       opoo <<-EOS.undent
         The git repository is a shallow clone therefore the filtering may be incorrect.
         Use `git fetch --unshallow` to get the full repository.
       EOS
     end
+
+    if ARGV.size > 2 then raise <<-EOS.undent
```

Comment:
Unfortunately, this will make a lot of legitimate use cases fail, because that's checking the length of the entire argument list (including options), e.g. `brew log --oneline ffmpeg 2.8.1`.